test/testAdaptors/AuraERC4626Adaptor.t.sol

```diff
@@ -0,0 +1,298 @@
+
+        initialAssets = cellar.totalAssets();
+    }
+
+    // deposit test: ensure that cellar deposit leads to transferance of BPT to aura pool. Cellar should get back aura-vault or aura-pool tokens back as receipts/shares to the auraPool.
+    function testDeposit(uint256 assets) external {
+        assets = bound(assets, 0.1e18, 1_000_000_000e18);
+        deal(address(rETH_wETH_BPT), address(this), assets);
+        cellar.deposit(assets, address(this));
+
+        uint256 assetsInAuraPool = auraRETHWETHBPTVault.balanceOf(address(cellar)); // as per Aura BaseRewardPool4626.sol - this is ASSUMING 1:1 for BPT (vault asset) : AuraBPT (shareToken)
+        // It is seen in the auraPool code that balanceOf() is overridden: ie. function balanceOf(address account) public view override(BaseRewardPool, IERC20) --> this means that it just reports how much `asset` (aka BPT) is in the auraPool for the user.
+
+        assertEq(assetsInAuraPool, assets, "Assets should have been deposited into assetsInAuraPool 1:1.");
+        assertEq(
+            cellar.totalAssets(),
+            initialAssets + assets,
+            "Cellar totalAssets should equal assets + initial assets, 1:1"
+        );
+    }
+
+    // withdraw test: ensure that cellar withdraw leads to transferance of BPT from aura pool back to cellar. Cellar should get back BPTs and have their aura receipts burnt.
+    function testWithdraw(uint256 assets) external {
+        assets = bound(assets, 0.1e18, 1_000_000_000e18);
+        deal(address(rETH_wETH_BPT), address(this), assets);
+        cellar.deposit(assets, address(this)); // bpts sent to aura pool as seen in deposit() tests
+
+        uint256 maxRedeem = cellar.maxRedeem(address(this));
+        uint256 redeemedAssets = cellar.redeem(maxRedeem, address(this), address(this)); // state mutative call to redeem assets to address(this)
+
+        assertApproxEqAbs(
+            assets,
+            redeemedAssets,
+            2,
+            "Time passing should not change the amount of BPTs sent to user unless strategist loops rewards to buy more BPTs or something else."
+        );
+        assertApproxEqAbs(
+            rETH_wETH_BPT.balanceOf(address(this)),
+            assets,
+            1,
+            "All BPTs should have been returned to user."
+        );
+        assertApproxEqAbs(
+            auraRETHWETHBPTVault.balanceOf(address(cellar)),
+            0,
+            2,
+            "Cellar should have redeemed all vault aura share tokens."
+        );
+    }
+
+    // same as redeem test but with a long time within the vault. Shows that rewards are not the vault asset (BPT) but other tokens.
+    function testFastForward(uint256 assets) external {
+        assets = bound(assets, 0.1e18, 1_000_000_000e18);
+        deal(address(rETH_wETH_BPT), address(this), assets);
+        cellar.deposit(assets, address(this)); // bpts sent to aura pool as seen in deposit() tests
+
+        vm.warp(block.timestamp + 100 days);
```

Comment:
I recently discovered you can use `skip(100 days)` instead if you want.

Reply:
That seems cleaner. Just tried it and it works!